### ISSUE TITLE
Platform vs System

### ISSUE THREAD

**Issue by EttoreHector** [urbanobservatory/standards#12]
I have some doubts about the use of "platform" vs "system".

The definitions are the following:   

Platform: This can be anything from a lamppost or tripod, through to a mobile platform such as a car, van or person. Systems and Sensors are hosted on these Platforms. The Platforms are recorded in a database collection called platforms.

System: A System is a group of Sensors (possibly also actuators) that implement a Procedure. A System is hosted by (isHostedBy) a Platform. An example of System is a weather station, a group of Sensors incorporated into an integrated unit which takes readings at regular intervals. 

For what I understand, Platform and System can effectively be the same thing. For example, a  fixed DEFRA air-quality station is both a Platform and a System. How shall we treat these cases?

**Comment by SiBell** [comment]
So this is how I've been doing it (which may not be right):

As yet I haven't needed to use a *system*. I have _sensors_ hosted on  _platforms_, and these _platforms_ can themselves be hosted on another _platforms_ .

So using your example: a NOx probe would be a _sensor_ (id: _defra-aq-number4-nox_), which is hosted on the DEFRA air-quality station (id: _defra-aq-number4_) which is a _platform_ because it hosts sensors. E.g. in the [SSN Docs](https://www.w3.org/TR/vocab-ssn/#iphone_barometer-sosa) they represent an iPhone7 as a platform. This air quality platform could then be hosted on another _platform_ called _london-road-lamppost-07_.

Now let's say when the air quality gets really bad an LED display (id: _led-display-serial-355253_) on the lamppost starts flashing a warning message to pedestrians. This process would then be called a *Procedure* and thus you could then use a *system* to group the *sensor* _defra-aq-number4-nox_ with the _actuator_  _led-display-serial-355253_. By grouping these as a *system* it sets them apart from any other _sensors/actuators_ on that lamppost that aren't involved in this *procedure*.

This is my take on it anyway.

**Comment by EttoreHector** [comment]
Thank you, Simon.

That clarifies things quite a lot.

**Comment by lukeshope** [comment]
I agree with Simon on the definitions. A `Sensor` or `Actuator` are subclasses of `System`, so in our cases we will always have at least one instance of a system. A system can have subsystems for the case Simon describes, where the system as a whole has a procedure to trigger a sign, which may be dependent on multiple sensors (that are subsystems).

**Comment by EttoreHector** [comment]
So we will always have a sensor as part of a System, which is always
eventually part of a Platform, which is always part of a Deployment.

**Comment by SiBell** [comment]
I'd say a _sensor_ doesn't HAVE to be part of a _system_. It's confusing because _sensors/actuators_ are _sub-classes_ of a system, but that doesn't mean a sensor/actuator HAS to belong to a _system_, they just have similar properties to a _system_.

**Comment by lukeshope** [comment]
It is confusing. I'm saying a sensor/actuator **is** a system (because it is a subclass), and therefore it is fine to associate sensors directly to platforms.

You can of course have more systems in between your sensors and your platforms, if you so desire.

**Comment by SiBell** [comment]
I'm going to throw out another example and let's see if we all agree.

Let's say a manufacturer makes an air quality protection device comprising off 3 _sensors_:

- An air quality (e.g. NO2) sensor.
- A passive infrared sensor that detects pedestrians nearby.
- A battery voltage sensor.

And 2 _actuators_:

- A servo that releases face masks to passers by when the air quality is dangerously high.
- A red LED that blinks when the device's battery is low.

So each of these 5 _sensors_/_actuators_ is a _system_, in that _sensors_ and _actuators_ are a sub-class of _system_.

All 5 _sensors_ are hosted by the "air quality protection device" as a whole, which is a _platform_.

Now the air quality sensor, the passive infrared sensor, and the servo are also part of their own _system_ running a _procedure_ (i.e. release a face mask when air quality is bad and person is nearby).

The battery voltage sensor and the red LED are part of a separate _system_ whose _procedure_ is to flash the LED when the battery is low.

Any objections to this? Happy to be proved wrong.

**Comment by lukeshope** [comment]
No objections, I agree with all of that.

**Comment by geoanorak** [comment]
I love these discussions although only lurking as not qualified to comment. But remember our principles should be about good enough for now with a little nod to foreseeable issues. Simplicity and usability above all. And I'm not saying that anything said so far strays off this path but wherever possible apply Occam's razor. 
Phil

Sent from Yahoo Mail on Android 
 
  On Wed, 14 Aug 2019 at 15:34, Luke Smith<notifications@github.com> wrote:   
No objections, I agree with all of that.

**Comment by EttoreHector** [comment]
Thank you again.

Yes, I agree with all of that.
